lib/pbench/server/api/resources/query_apis/metadata_index/index_mappings.py

```diff
@@ -59,16 +59,22 @@ class IndexMappings(Resource):
     def __init__(self, logger: Logger):
         self.logger = logger
 
-    def get(self, index_name: AnyStr) -> Dict[AnyStr, Any]:
+    def get(self, index_key: AnyStr) -> Dict[AnyStr, Any]:
         """
-        Return mapping properties of the document specified by the index name in the URI.
-        For example, user can get the run document properties by making a GET request on
-        index/mappings/run. Similarly other index documents can be fetched by making a GET
-        request on appropriate index names.
-        We fetch the mapping by querying the template database. If the template is not found
-        in the database NOT_FOUND error will be raised.
+        Return mapping properties of the document specified by the index name
+        in the URI. For example, user can get the run document properties by
+        making a GET request on index/mappings/search. Similarly other index
+        documents can be fetched by making a GET request on appropriate index names.
+
+        We fetch the mapping by querying the template database. If the
+        template is not found in the database NOT_FOUND error will be raised.
```

Comment:
The user input parameter no longer directly corresponds to a "template" but rather to one of the keywords you've defined in `ES_INTERNAL_INDEX_NAMES`; so this is really "If the keyword isn't valid". I'm not entirely sure I like `NOT_FOUND` anymore, either, since we're really not representing a resource directly... I think I'd rather have it look like a keyword validation error (which it is) as we'd get from a schema validation.

In fact, you could do this by defining a class `Parameter` object using your `ES_INTERNAL_INDEX_NAMES` as the keyword list, and validate against that. This is more or less what I did in the `get` for `datasets_metadata.py` to validate URI parameter keywords. (Look at the definition and use of `GET_SCHEMA`.)

Reply:
And I should also probably use the ApiBase to inherit instead of direct flask Resource class